src/Admin/AbstractAdmin.php

```diff
@@ -849,13 +848,40 @@ public function buildDatagrid()
      * Returns the name of the parent related field, so the field can be use to set the default
      * value (ie the parent object) or to filter the object.
      *
-     * @return string the name of the parent related field
+     * @throws \InvalidArgumentException
+     *
+     * @return null|string
      */
     public function getParentAssociationMapping()
     {
+        // NEXT_MAJOR: remove array check
+        if (\is_array($this->parentAssociationMapping) && $this->getParent()) {
+            $parent = $this->getParent()->getCode();
```

Comment:
How would `$this->getParent()` know which parent to call? I think this needs more investigating